public/app/features/serviceaccounts/ServiceAccountsListPage.tsx

```diff
@@ -72,10 +72,12 @@ const ServiceAccountsListPage = ({
     }
   }, [fetchServiceAccounts, fetchACOptions]);
 
-  const onRoleChange = (role: OrgRole, serviceAccount: ServiceAccountDTO) => {
+  const onRoleChange = async (role: OrgRole, serviceAccount: ServiceAccountDTO) => {
     const updatedServiceAccount = { ...serviceAccount, role: role };
-    updateServiceAccount(updatedServiceAccount);
+    await updateServiceAccount(updatedServiceAccount);
+    await fetchServiceAccounts();
```

Comment:
That's also might be triggered from actions, but I'm not sure we need to update service accounts list every time we change single account.